src/fr/charly/miniCalculator/Calculator.java

```diff
@@ -51,16 +51,42 @@ public Calculator() {
 	private class ButtonClickListener implements ActionListener {
```

Comment:
Can you please create this class apart from this class? This is a one of the best practices in java. No class in another class.